### DISCUSSION TITLE
Why it doesn't recognize refs and sources?

### DISCUSSION THREAD

**Discussion by pawanyadavnitk** [AltimateAI/vscode-dbt-power-user#159] [General] [answered]
The extension recognizes refs in one project but doesn't recognize them in another.

What are the possible reasons? Can anyone please help?

Might be related to the folder structure or what?

**Comment by mdesmet** [comment]
Could you explain or investigate what are the differences between the project that works and the project that doesn't work?

**Reply by pawanyadavnitk** [reply]
The folder structure is such that it has two dbt_project.yml flies. One is in the root directory and the other is inside a child directory. I want to use the dbt project in the child folder but keep the VSCode open in the parent folder.

Because there are some other files and folders outside of my child's dbt folder that is important to me.

The question can be reframed as: How do I tell the extension that this is the dbt_project.yml file that I want to use? And that shouldn't necessarily be present in the root folder in which VSCode is open.

**Reply by mdesmet** [reply]
The folder structures must always be valid dbt projects. So just putting a dbt_project.yml file in a child directory is probably causing issues. why not put them both in separate child directories.

**Reply by pawanyadavnitk** [reply]
It's a valid dbt project. But there is a different reason that two dbt_project.yml files are present, and I don't have any control over that.
 
How do I use the extension if my dbt_project.yml file is not in the root directory? This is what I want to know, can you please answer this?

**Reply by mdesmet** [reply]
A setup with multiple dbt projects is supported but you say you have a dbt_project.yml in the root directory and one in a child directory. That would give unexpected results. I would only expect other dbt projects to be present in a `dbt_packages` folder. Just to be clear, the extension supports placing your projects where ever you want in a vscode workspace, it just discovers them based on finding all `dbt_project.yml` files.

**Comment by pawanyadavnitk** [comment]
Could you please help me with this on a call? I can schedule a short 10 min call according to your availability. Thanks

**Comment by mdesmet** [comment, marked as answer]
You can talk to me on dbt Slack.
